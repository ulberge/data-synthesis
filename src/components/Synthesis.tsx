import React, { useState, useCallback } from 'react';
import Assets from './Assets';

/**
 * Component that renders and manages the UI workflow for data synthesis.
 */
function Synthesis() {
  // List of image assets loaded through drag and drop area
  const [ assets, setAssets ] = useState<File[]>([]);
  const [ selectedAsset, setSelectedAsset ] = useState<File | undefined>();

  return (
    <div>
      <Assets assets={assets} setAssets={setAssets} selectedAsset={selectedAsset} setSelectedAsset={setSelectedAsset} />
    </div>
  );
}

export default Synthesis;
