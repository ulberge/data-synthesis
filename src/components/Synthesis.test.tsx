import React from 'react';
import { render, waitForElement } from '@testing-library/react';
import Synthesis from './Synthesis';

test('renders Synthesis component', () => {
  const { container } = render(<Synthesis />);
  expect(container).toMatchSnapshot();
});
