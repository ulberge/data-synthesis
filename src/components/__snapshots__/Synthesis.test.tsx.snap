// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Synthesis component 1`] = `
<div>
  <div>
    <div
      class="dragdrop"
      tabindex="0"
    >
      <p
        class="center"
      >
        Drop assets here
      </p>
    </div>
  </div>
</div>
`;
